src/modules/0.0.14/aws_iam/index.ts

```diff
@@ -180,6 +180,10 @@ export const AwsIamModule: Module2 = new Module2({
             const newRole = await AwsIamModule.utils.roleMapper(rawRole, ctx);
             await AwsIamModule.mappers.role.db.update(newRole, ctx);
             out.push(newRole);
+            // TODO: find a way to fix this better?
+            // wait 10 seconds to let aws get the role ready to use :/
+            // https://stackoverflow.com/a/37438525
+            await new Promise(r => setTimeout(r, 10000));
```

Comment:
:shrug:

Reply:
horrible I know. Not sure how to fix it properly, it also increases exponentially IAM tests run time. I'll try to set a lower number and see what happens :/

Reply:
Is there a waiter to wait for the IAM role to be ready to use?

Reply:
https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-iam/globals.html#waitforroleexists Maybe this would work?

Reply:
The issue is that the role in question seems to be defined, but the error is `The role defined for the function cannot be assumed by Lambda`. The assume role part is done by passing the `AssumeRolePolicyDocument` property creating the role. I've seen that waiter and another for policies, I'll try to run both of thems and see if that works.

Reply:
Nope, the waiter does not work and I've found this https://stackoverflow.com/questions/20156043/how-long-should-i-wait-after-applying-an-aws-iam-policy-before-it-is-valid. Apparently there's no way no know this :/

Reply:
Alternative solution. In the Lambda module, this is a "known" possible error, and just have an internal try-catch-sleep loop for up to say, 10 iterations with 5 second sleeps?

Reply:
oh definitely better and do not impact IAM module